app/models/user.rb

```diff
@@ -21,6 +24,18 @@ def video_in_user_queue?(video_id)
     self.queue_items.pluck(:video_id).include?(video_id)
   end
 
+  def follows?(another_user)
+    self.following_relationships.map(&:leader).include?(another_user)
+  end
+
+  def can_follow?(another_user)
+    !(self.follows?(another_user) || self == another_user)
+  end
+
+  def full_name
+    full_name = self.first_name + " " + self.last_name
```

Comment:
It might be your own preference, but we comment based on style guide that is generally accepted by Ruby community.